queries.sql

```diff
@@ -129,4 +129,70 @@ FROM animals
 JOIN owners ON animals.owner_id = owners.id 
 GROUP BY owners.full_name 
 ORDER BY COUNT(*) DESC 
-LIMIT 1;
\ No newline at end of file
+LIMIT 1;
+
+/*add "join table" for visits queries*/
+
+SELECT a.name 
+FROM animals a 
+JOIN visits v ON a.id = v.animal_id 
+JOIN vets vt ON v.vet_id = vt.id 
+WHERE vt.name = 'William Tatcher' 
+ORDER BY v.visit_date DESC 
+LIMIT 1;
+
+SELECT COUNT(DISTINCT v.animal_id) 
+FROM visits v 
+JOIN vets vt ON v.vet_id = vt.id 
+WHERE vt.name = 'Stephanie Mendez';
+
+SELECT vt.name, s.name AS specialty
+FROM vets vt
+LEFT JOIN specializations sp ON vt.id = sp.vet_id
+LEFT JOIN species s ON s.id = sp.species_id
+ORDER BY vt.name;
+
+SELECT a.name 
+FROM animals a 
+JOIN visits v ON a.id = v.animal_id 
+JOIN vets vt ON v.vet_id = vt.id 
+WHERE vt.name = 'Stephanie Mendez' 
+AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+SELECT a.name AS animal_name, COUNT(*) AS visit_count 
+FROM animals a 
+JOIN visits v ON a.id = v.animal_id 
+GROUP BY a.id 
+ORDER BY visit_count DESC 
+LIMIT 1;
+
+SELECT a.name AS animal_name, MIN(v.visit_date) AS first_visit
+FROM animals a
+INNER JOIN visits v ON a.id = v.animal_id
+INNER JOIN vets vt ON vt.id = v.vet_id
+WHERE vt.name = 'Maisy Smith'
+GROUP BY a.name;
+
+SELECT a.*, v.*, MAX(visits.visit_date) AS most_recent_visit_date
+FROM visits
+JOIN animals a ON visits.animal_id = a.id
+JOIN vets v ON visits.vet_id = v.id
+WHERE visits.visit_date = (SELECT MAX(visit_date) FROM visits)
+GROUP BY a.id, v.id
+
+SELECT COUNT(*)
+FROM visits v
+INNER JOIN animals a ON a.id = v.animal_id
+INNER JOIN vets vt ON vt.id = v.vet_id
+LEFT JOIN specializations s ON vt.id = s.vet_id AND a.species_id = s.species_id
+WHERE s.vet_id IS NULL;
+
+SELECT a.species_id, COUNT(*) AS num_visits
+FROM visits v
+INNER JOIN animals a ON v.animal_id = a.id
+WHERE v.vet_id = (
+  SELECT id FROM vets WHERE name = 'Maisy Smith'
+)
+GROUP BY a.species_id
```

Comment:
- [ ] @Leeoasis great work on your queries however I would recommend you group this query by species name instead of the id, this will make it easier to identify the species that Maisy Smith should consider getting. Don't forget to update the query result screenshot